pkg/internal/sqlsmith/relational.go

```diff
@@ -537,20 +534,24 @@ func makeSelectTable(s *Smither, refs colRefs, forJoin bool) (tree.TableExpr, co
 
 func makeSelectClause(
 	s *Smither, desiredTypes []*types.T, refs colRefs, withTables tableRefs,
-) (tree.SelectStatement, colRefs, tableRefs, bool) {
-	stmt, selectRefs, _, tables, ok := s.makeSelectClause(desiredTypes, refs, withTables)
-	return stmt, selectRefs, tables, ok
+) (tree.SelectStatement, colRefs, bool) {
+	stmt, selectRefs, _, ok := s.makeSelectClause(desiredTypes, refs, withTables)
+	return stmt, selectRefs, ok
 }
 
 func (s *Smither) makeSelectClause(
 	desiredTypes []*types.T, refs colRefs, withTables tableRefs,
-) (clause *tree.SelectClause, selectRefs, orderByRefs colRefs, tables tableRefs, ok bool) {
+) (clause *tree.SelectClause, selectRefs, orderByRefs colRefs, ok bool) {
 	if desiredTypes == nil && s.d9() == 1 {
 		return s.makeOrderedAggregate()
 	}
 
 	clause = &tree.SelectClause{}
 
+	s.rnd.Shuffle(len(withTables), func(i, j int) {
+		withTables[i], withTables[j] = withTables[j], withTables[i]
```

Comment:
This makes it so that each SELECT can use a WITH table once, but if there are multiple SELECTs both can use the same WITH table. Instead, I'd like a single SELECT to be able to use the same WITH table more than one time. So this shuffle can be removed. The withTables.Pop down below should be removed and instead it just picks a random table. I think it'd be:

```
table := withTables[s.rng.Intn(len(withTables))]
```

Reply:
Done :)

Reply:
Done